src/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegment.cs

```diff
@@ -78,6 +78,31 @@ public void ResetMemory()
 
         public Memory<byte> AvailableMemory { get; private set; }
 
+        public IMemoryList<byte> GetNext(long offset, out int localIndex)
+        {
+            BufferSegment current = this;
+            while (current != null)
+            {
+                var currentLength = current.Memory.Length;
+                if (offset <= currentLength)
```

Comment:
Could this be something like `offset < currentLength` to try and support the scenario within SeekMultiSegment inside this method?

https://github.com/dotnet/corefx/pull/27436/files#diff-6a3b2bb4fe535f3ac47622b764877747R180

```C#
if (localOffset == memoryList.Memory.Length && memoryList != end)
```

Reply:
No, `IMemoryList` has to be conservative because it doesn't know where ReadOnlySegment ends. Note the `memoryList != end` check in `ReadOnlySegments` logic that makes sure we don't cross outside the bounds.